workflow/energy_rating_index.rb

```diff
@@ -149,16 +148,20 @@ def read_output(design, designdir, output_hpxml_path)
     query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue IN (#{keys}) AND VariableName IN (#{vars}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
     design_output[:fuelHeatingBySystem][sys_id] = get_sql_query_result(sqlFile, query)
-    # Load
-    vars = "'" + get_all_var_keys(OutputVars.SpaceHeatingLoad).join("','") + "'"
-    query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue IN (#{keys}) AND VariableName IN (#{vars}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
-    design_output[:loadHeatingBySystem][sys_id] = get_sql_query_result(sqlFile, query)
     # Disaggregated Fan Energy Use
-    keys = "'" + ep_output_names.select { |name| name.include? "Heating" }.join("','") + "'"
-    query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue='EMS' AND VariableName IN (#{keys}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
+    ems_keys = "'" + ep_output_names.select { |name| name.include? "Heating" }.join("','") + "'"
+    query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue='EMS' AND VariableName IN (#{ems_keys}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
     fan_pump_output = get_sql_query_result(sqlFile, query)
     design_output[:elecHeatingBySystem][sys_id] += fan_pump_output
-    design_output[:loadHeatingBySystem][sys_id] += fan_pump_output
+    # Reference Load
+    if [Constants.CalcTypeERIReferenceHome, Constants.CalcTypeERIIndexAdjustmentReferenceHome].include? design
+      # Only ever gas furnace, gas boiler, or electric ASHP (autosized)
+      vars = "'" + get_all_var_keys(OutputVars.SpaceHeatingLoad).join("','") + "'"
+      query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue IN (#{keys}) AND VariableName IN (#{vars}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
+      design_output[:loadHeatingBySystem][sys_id] = get_sql_query_result(sqlFile, query)
+      design_output[:loadHeatingBySystem][sys_id] += fan_pump_output
+      design_output[:loadHeatingBySystem][sys_id] *= design_output[:hpxml_dse_heats][sys_id] # Remove effect of DSE on load
```

Comment:
Basic changes:
1. Added comment that this is the _Reference_ Load and which HVAC system types this applies to.
2. Only perform calculations for reference desgins.
3. Moved all calculations into the same code block (e.g., DSE, fan_pump_output).

Reply:
I have one concern for indirect water system, because we would want the heating load for rated home which is used to calculate indirect fraction. And the heating load of reference home is different from that of rated home. Maybe I could try adding boiler object to rated home indirect water heating map and read output variables there.

Reply:
I'm not trying to handle indirect water heater here, just improving current capabilities. I'm assuming we'll merge this in, then you'll pull it into your combi system branch and handle those systems.

Reply:
Yes I mean I am assuming it is merged and might need to add some new objects to dhw maps. 😄